opal/mca/base/mca_base_alias.c

```diff
@@ -93,13 +93,13 @@ static char *mca_base_alias_generate_name (const char *project, const char *fram
 
     if (project_length) {
         strncat (tmp, project, length);
-        strncat (tmp, "_", 1);
```

Comment:
I agree here. In this case `strcat(tmp, "_")` would imo be sufficient since we have full control over the source parameter so there is no need to specify the number of bytes to copy.

Reply:
I remember thinking that the compiler warning I was getting here was quite odd -- it was something about how the destination was exactly the same size as the source.  However, I'm now unable to replicate that compiler warning.  So I'll just drop this change.

Reply:
Ah, wait, I found this warning again.  The warning is happening on GCC 10 on Linux:

```
mca_base_alias.c: In function ‘mca_base_alias_generate_name’:
mca_base_alias.c:96:9: warning: ‘strncat’ specified bound 1 equals source length [-Wstringop-overflow=]
   96 |         strncat (tmp, "_", 1);
      |         ^~~~~~~~~~~~~~~~~~~~~
mca_base_alias.c:102:9: warning: ‘strncat’ specified bound 1 equals source length [-Wstringop-overflow=]
  102 |         strncat (tmp, "_", 1);
      |         ^~~~~~~~~~~~~~~~~~~~~
```

Googling around and thinking about the message a bit, I think it's trying to tell us that calling `strncat()` with a constant source string of length 1 and *also* passing in a source length of 1 is the equivalent of calling `strcat()`.  I.e., we didn't really gain anything from calling `strncat()` here.

You may or may not agree with GCC's warning here, but... how should we eliminate it?

1. Set the source length to 2 (which is harmless).
    * This feels like wankery just to avoid a compiler warning, and makes the code a little... weird.
1. Use `strcat()`.
    * This might be the least-bad answer here.  With a constant source string, it _is_ the same thing as calling `strcat()`.

How about I convert to `strcat()` and put in a comment explaining why?

Reply:
Let's go with strcat, looks simpler and less confusing to people (inline passing 2 as the number of bytes when the static char* is of length 1).